### PR TITLE
Fix remedy on demand fetch error

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/43926

## Description
Fixed the query of fetch incidents

### CODE REVIEW

**Review by reutshal — APPROVED** [review]

File: Packs/Remedy-On-Demand/ReleaseNotes/1_0_3.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Remedy On-Demand
+- Fixed an issue with the query of ***fetch-incidents***.
```
Comment (amshamah419) [inline Packs/Remedy-On-Demand/ReleaseNotes/1_0_3.md:4]:
```suggestion
- Fixed an issue where the query of the ***fetch-incidents*** command was not formatted correctly.
```

**Comment by reutshal** [comment]
@MosheEichler When was the call scheduled? we are waiting on this or we are sure about the fix?

**Comment by MosheEichler** [comment]
We are sure about the fix (as the customerד mentioned in the [issue](https://github.com/demisto/etc/issues/43926#issuecomment-985834668) that the fetch works now)
But we are waiting for a call with the customers about duplicates in fetch incidents (could be related to the fix)
So let's wait with the merge of this pr untill call with the customers

File: Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js [hunk 2]
```diff
@@ -210,15 +210,15 @@ var updateIncident = function(incID, updateObject) {
...
-    var now = nowDate.toISOString();
     if (!lastRun || !lastRun.value) {
         lastRun = {
-            value: (new Date(nowDate.getTime() - 10*60*1000)).toISOString()
+            value: (new Date(new Date().getTime() - 10*60*1000)).toISOString()
         };
     }
-    var query =  "'Create Date'>" + '"' + lastRun.value + '"';
-    var url = baseUrl + '/api/arsys/v1/entry/HPD:IncidentInterface/' + '?q=' + encodeURIComponent(query);
+    logDebug("This is the lastrun value before starting to fetch: "+lastRun.value);
```
Comment (reutshal) [inline Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js:218]:
```suggestion
    logDebug("Last run value before starting to fetch: "+lastRun.value);
```

File: Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js [hunk 3]
```diff
@@ -238,6 +238,8 @@ var fetchIncidentsToDemisto = function() {
             'rawJSON': JSON.stringify(incident)
         });
     });
+    var now = new Date().toISOString();
+    logDebug("it's the time now after the fetch, it's saved for the next fetch: "+now);
```
Comment (reutshal) [inline Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js:242]:
```suggestion
    logDebug("Last run is set to: "+now);
```

File: Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js [hunk 4]
```diff
@@ -210,15 +210,15 @@ var updateIncident = function(incID, updateObject) {
...
         lastRun = {
-            value: (new Date(nowDate.getTime() - 10*60*1000)).toISOString()
+            value: (new Date(new Date().getTime() - 10*60*1000)).toISOString()
         };
     }
-    var query =  "'Create Date'>" + '"' + lastRun.value + '"';
-    var url = baseUrl + '/api/arsys/v1/entry/HPD:IncidentInterface/' + '?q=' + encodeURIComponent(query);
+    logDebug("This is the lastrun value before starting to fetch: "+lastRun.value);
+    var query =  '\'Create Date\'>' + '\"' + lastRun.value + '\"';
+    var url = baseUrl + '/api/arsys/v1/entry/HPD:IncidentInterface/' + '?q=' + query;
```
Comment (reutshal) [inline Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js:220]: Why did we remove the encoding to the query?

**Review by reutshal — CHANGES_REQUESTED** [review]

File: Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js [hunk 5]
```diff
@@ -219,8 +219,10 @@ var fetchIncidentsToDemisto = function() {
             value: (new Date(nowDate.getTime() - 10*60*1000)).toISOString()
         };
     }
-    var query =  "'Create Date'>" + '"' + lastRun.value + '"';
+    logDebug("Last run value before starting to fetch: "+lastRun.value);
+    var query =  "'Submit Date'>" + '"' + lastRun.value + '"';
     var url = baseUrl + '/api/arsys/v1/entry/HPD:IncidentInterface/' + '?q=' + encodeURIComponent(query);
+    logDebug("This is the URL with the query for fetching the incidents: "+url);
```
Comment (reutshal) [inline Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js:225]:
```suggestion
    logDebug("This is the URL with the query for fetching the incidents: " + url);
```

File: Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js [hunk 6]
```diff
@@ -219,8 +219,10 @@ var fetchIncidentsToDemisto = function() {
             value: (new Date(nowDate.getTime() - 10*60*1000)).toISOString()
         };
     }
-    var query =  "'Create Date'>" + '"' + lastRun.value + '"';
+    logDebug("Last run value before starting to fetch: "+lastRun.value);
```
Comment (reutshal) [inline Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js:222]:
```suggestion
    logDebug("Last run value before starting to fetch: " + lastRun.value);
```

File: Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js [hunk 7]
```diff
@@ -240,6 +242,8 @@ var fetchIncidentsToDemisto = function() {
             'rawJSON': JSON.stringify(incident)
         });
     });
+    var now = new Date().toISOString();
+    logDebug("Last run is set to: "+now);
```
Comment (reutshal) [inline Packs/Remedy-On-Demand/Integrations/RemedyOnDemand/RemedyOnDemand.js:246]:
```suggestion
    logDebug("Last run is set to: " + now);
```

**Review by reutshal — APPROVED** [review]

**Pull request merged**